src/opcodes/tests/test_stack.cairo

```diff
@@ -290,3 +290,22 @@ fn test_op_2rot_insufficient_items() {
     let mut engine = utils::test_compile_and_run_err(program, Error::STACK_OUT_OF_RANGE);
     utils::check_dstack_size(ref engine, 5);
 }
+
+#[test]
+fn test_opcode_over() {
+    let program = "OP_OVER";
```

Comment:
You'll need to push values onto the stack for op_over to use. I'd recommend using the examples on this page https://en.bitcoin.it/wiki/Script

Reply:
> You'll need to push values onto the stack for op_over to use. I'd recommend using the examples on this page https://en.bitcoin.it/wiki/Script

the script link specify we pass two value (i.e x1 x2) and expect  x1 x2 x1 as output,

which is what i did here 

```
#[test]
fn test_opcode_over() {
    let program = "OP_1 OP_2 OP_OVER";
    let mut engine = utils::test_compile_and_run(program);
    utils::check_dstack_size(ref engine, 3);
    let expected_dstack = array![ScriptNum::wrap(1), ScriptNum::wrap(2), ScriptNum::wrap(1)];
    utils::check_expected_dstack(ref engine, expected_dstack.span());
}
```

i still got the following error,


`
shinigami::opcodes::tests::test_stack::test_opcode_over - Panicked with "Execution of the program failed".
`

could the error came as a result of opcode_over function failing to run successfully

can you inspect the function and let me know if this

Reply:
I'd recommend using println! statements to debug. Your test is now setup correct.
Debugging could be good practice, let me know.

Reply:
Alright

Reply:
println!("{:?}", bytecode);
    let mut engine = EngineTraitImpl::new(bytecode);
    let res = engine.execute();
    println!("{:?}", res);
    
    return the following result 
    
```
    
"QR\0"
Result::Err(34304024605757925039519833781264315913130327289331821709226243173)
```

this is the opcode program

"OP_1 OP_2 OP_OVER"

Reply:
Could you use println! within the over_n implementations to see where the issue is occurring

Reply:
the code does not reach that point,

both opcode_over and over_n.

Reply:
I think you never added the opcodes to the compiler

Reply:
hmm

you are right

Reply:
i suspect the while loop am using

Reply:
test from function [test_opcode_over] is running successfully now,

Reply:
test_opcode_2over  is also running successfully